pkg/mount/dag_steps.go

```diff
@@ -176,24 +176,18 @@ func (s *State) LoadEnvLayoutDagStep(g *herd.Graph, opts ...herd.OpOption) error
 func (s *State) MountOemDagStep(g *herd.Graph, opts ...herd.OpOption) error {
 	return g.Add(cnst.OpMountOEM,
 		append(opts,
-			herd.EnableIf(func() bool {
-				runtime, _ := state.NewRuntime()
-				switch runtime.BootState {
-				// Don't run this on LiveCD/Netboot
-				case state.LiveCD:
-					return false
-				default:
-					return internalUtils.GetOemLabel() != ""
-				}
-			}),
 			herd.WithCallback(func(ctx context.Context) error {
-				// We have to run the check here because otherwise is run on start instead of when we want to mount oem
-				// And at program start we have not mounted the efivarsfs so this would always return false
-				if internalUtils.IsUKI() {
-					if !internalUtils.EfiBootFromInstall() {
-						return nil
-					}
+				runtime, _ := state.NewRuntimeWithLogger(internalUtils.Log)
+				if runtime.BootState == state.LiveCD {
+					internalUtils.Log.Debug().Msg("Livecd mode detected, won't mount OEM")
+					return nil
 				}
+				// TODO: when does this happen?
```

Comment:
@Itxaka do you know? we can put is as a comment here

Reply:
when what happens exactly?

the empty label? I have no idea when that can happen but someone could modify the cmdline to not provide that oem and that results into a very weird bug on where it tries to mount an empty oem if it snot give anywhere (not sure hwy we let people choose the oem label, we should deprecate that) In that case, instead of blocking the boot and then failing but still booting, we just log this and skip trying to mount so it doesnt get stuck retrying to mount the oem.

Reply:
ok, so the log messages pretty much describes what happened. Someone messed with the cmdline. I will remove the TODO, thanks.